packages/side-nav/src/SideNavGroupCollapsed/SideNavGroupCollapsed.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect } from 'react';
+import { Transition, TransitionStatus } from 'react-transition-group';
+
+import { css, cx } from '@leafygreen-ui/emotion';
+import { useIdAllocator } from '@leafygreen-ui/hooks';
+import ChevronRight from '@leafygreen-ui/icon/dist/ChevronRight';
+import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { palette } from '@leafygreen-ui/palette';
+
+import { useSideNavContext } from '../SideNav/SideNavContext';
+import { ulStyleOverrides } from '../SideNav/styles';
+import {
+  baseStyle,
+  buttonClassName,
+  indentedStyle,
+  themeStyle,
+} from '../SideNavGroup/styles';
+
+import {
+  collapsibleBaseStyle,
+  collapsibleFocusStyle,
+  collapsibleGroupBaseStyles,
+  collapsibleThemeStyle,
+  expandIconStyle,
+  openExpandIconStyle,
+  transitionStyles,
+  ulStyles,
+  ulTransitionStyles,
+} from './styles';
+import { SideNavGroupCollapsedProps } from './types';
+
+/**
+ * @internal
+ */
+export function SideNavGroupCollapsed({
+  groupHeaderProps,
+  open,
+  setOpen,
+  menuGroupLabelId,
+  indentLevel,
+  renderHeader,
+  children,
+}: SideNavGroupCollapsedProps) {
+  const { width, theme, darkMode } = useSideNavContext();
+  const { usingKeyboard } = useUsingKeyboardContext();
+
+  const menuId = useIdAllocator({ prefix: 'menu' });
+
+  const nodeRef = React.useRef(null);
+  const ulRef = React.useRef<HTMLUListElement>(null);
+
+  // compute the entered ul wrapper styles based on the ul height
+  useEffect(() => {
+    const ulHeight = ulRef?.current?.getBoundingClientRect().height ?? 0;
+    transitionStyles['entered'] = css`
+      opacity: 1;
+      max-height: ${ulHeight + 1}px; // +1 for border
+      border-bottom: 1px solid
+        ${darkMode ? palette.gray.dark1 : palette.gray.light2};
+    `;
```

Comment:
Not sure if this is something you added in this PR, but it feels strange to me that you're changing styles in a useEffect. I think it would make more sense if you passed `ulHeight` as a parameter to a style function that's fully defined in the styles file.

Reply:
I'm going to leave this as is since I didn't add this.